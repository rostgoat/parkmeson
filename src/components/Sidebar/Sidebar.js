import React, { useContext, useState } from 'react'

import { Button, Text, Container, List, ListItem } from 'native-base'
import { StyleSheet, View, ScrollView } from 'react-native'

import { MeterContext } from '../../context/meterContext'
import { MeterFavContext } from '../../context/meterFavContext'

const MetersList = ({ meters, fav }) => {
  let tempMeters = []
  if (fav) {
    tempMeters = meters
  } else {
    tempMeters = meters.sort((a, b) => a.count - b.count)
  }

  return (
    <List>
      <ListItem itemHeader first>
        <Text>Pay By Phone</Text>
        {fav && <Text style={styles.countText}>Visits</Text>}
        <Text style={styles.countText}>Area</Text>
      </ListItem>
      {meters &&
        meters.length > 0 &&
        tempMeters.map((meter, index) => (
          <ListItem key={index}>
            <Text>{meter.id}</Text>
            {fav && <Text style={styles.count}>{meter.count}</Text>}
            <Text style={styles.area}>{meter.area}</Text>
          </ListItem>
        ))}
    </List>
  )
}
const Sidebar = () => {
  const [meters] = useContext(MeterContext)
  const [favMeters] = useContext(MeterFavContext)
  const [showFav, setShowFav] = useState(false)
  return (
    <Container style={styles.container}>
      <View style={styles.innerContainerButtons}>
        <Button rounded onPress={() => setShowFav(false)}>
          <Text>Most Used</Text>
        </Button>
        <Button rounded success onPress={() => setShowFav(true)}>
          <Text>Favorites</Text>
        </Button>
      </View>
      <ScrollView style={styles.innerContainerMeters}>
        {showFav && favMeters && favMeters.length > 0 && (
          <MetersList meters={favMeters} />
        )}
        {!showFav && meters && meters.length > 0 && (
          <MetersList meters={meters} fav={true} />
        )}
      </ScrollView>
    </Container>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    padding: 10,
    paddingTop: 30,
  },
  innerContainerButtons: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingLeft: 30,
    paddingRight: 30,
  },
  innerContainerMeters: {
    padding: 20,
  },
  count: {
    marginLeft: 70,
  },
  area: {
    marginLeft: 50,
  },
  countText: {
    marginLeft: 30,
  },
})

export default Sidebar
