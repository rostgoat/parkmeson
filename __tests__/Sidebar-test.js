/**
 * @format
 */

import 'react-native'
import React from 'react'

import renderer from 'react-test-renderer'

import { Sidebar } from 'components/Sidebar'

// Note: test renderer must be required after react-native.

it('renders correctly', () => {
  const tree = renderer.create(<Sidebar />).toJSON()
  expect(tree).toMatchSnapshot()
})
