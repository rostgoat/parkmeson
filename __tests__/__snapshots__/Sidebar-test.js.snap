// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "flex": 1,
        "padding": 10,
        "paddingTop": 30,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingLeft": 30,
        "paddingRight": 30,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 57,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        Most Used
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#5cb85c",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 57,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
      >
        Favorites
      </Text>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
